tools/rosbag/src/recorder.cpp

```diff
@@ -167,6 +168,23 @@ int Recorder::run() {
 
     start_time_ = ros::Time::now();
 
+    // Calculate a shift if we're splitting on time duration.  We want to
+    // end up splitting on "nice" wall clock values.  To do this we shift the
+    // start time by the time elapsed since the "previous" nice interval would
+    // have started.
+    if (options_.max_duration.toSec() > 0 && options_.split)
+    {
+        // Our last interval number
+        unsigned int prev_interval = (unsigned int) (floor(start_time_.toSec() / options_.max_duration.toSec()));
+
+        // This is the ros::Time that marks the beginning of the interval we've started recording in.
+        ros::Time prev_rollover_time;
+        prev_rollover_time.fromSec(prev_interval * options_.max_duration.toSec());
+
+        // Shift the start time backwards by our 'elapsed' time since the previous period begin
+        start_time_ -= (start_time_ - prev_rollover_time);
```

Comment:
This seems to be changing the `start_time` unconditionally of using the new option?

Reply:
This is fixed now, as the whole block is behind `if (options_.nice_split_times &&`

Reply:
Currently the value of `start_time_` is set to `prev_rollover_time` (because `start_time_ - start_time_` is always `0`). I would suggest to either change the assignment explicitly to that or instead compute a variable which contain the duration of `now` since the last interval start and subtract that variable from `start_time_`.